viper/src/verification_result.rs

```diff
@@ -70,3 +78,82 @@ pub struct JavaExceptionWithOrigin {
     /// The actual exception.
     pub exception: JavaException,
 }
+
+pub trait Cache {
+    fn get(&self, request: u64) -> Option<VerificationResult>;
+    fn insert(self, request: u64, result: VerificationResult) -> Option<VerificationResult>;
+}
+#[derive(Debug, Clone)]
+pub struct CacheData {
```

Comment:
What about `PersistentCache` instead of `CacheData`?

Also, I'd move all the cache logic to a new `src/cache.rs` file.